paddle/fluid/operators/bilinear_interp_op.cu.h

```diff
@@ -0,0 +1,101 @@
+                                   const size_t numChannels, const T ratioH,
+                                   const T ratioW) {
+  int nthreads = outputH * outputW;
+  int tid = blockIdx.x * blockDim.x + threadIdx.x;
+  if (tid < nthreads) {
+    int outIdH = tid / outputW;
+    int outIdW = tid % outputW;
+    int inImgSize = inputW / numChannels;
+    int outImgSize = outputW / numChannels;
+    int channelId = outIdW / outImgSize;
+
+    int outImgIdy = (outIdW % outImgSize) / outImgW;
+    int inImgIdy = ratioH * outImgIdy;
+    int hId = (inImgIdy < inImgH - 1) ? 1 : 0;
+    T h1lambda = ratioH * outImgIdy - inImgIdy;
+    T h2lambda = 1.f - h1lambda;
+
+    int outImgIdx = tid % outImgW;
+    int inImgIdx = ratioW * outImgIdx;
+    int wId = (inImgIdx < inImgW - 1) ? 1 : 0;
+    T w1lambda = ratioW * outImgIdx - inImgIdx;
+    T w2lambda = 1.f - w1lambda;
+
+    const T* inPos = &in[outIdH * inputW + channelId * inImgSize +
+                         inImgIdy * inImgW + inImgIdx];
+
+    // bilinear interpolation
+    out[outIdH * outputW + outIdW] =
+        h2lambda * (w2lambda * inPos[0] + w1lambda * inPos[wId]) +
+        h1lambda * (w2lambda * inPos[hId * inImgW] +
+                    w1lambda * inPos[hId * inImgW + wId]);
+  }
+}
+
+template <typename T>
+__global__ void KeBilinearInterpBw(T* in, const size_t inImgH,
+                                   const size_t inImgW, const size_t inputH,
+                                   const size_t inputW, const T* out,
+                                   const size_t outImgH, const size_t outImgW,
+                                   const size_t outputH, const size_t outputW,
+                                   const size_t numChannels, const T ratioH,
+                                   const T ratioW) {
+  int nthreads = outputH * outputW;
+  int tid = blockIdx.x * blockDim.x + threadIdx.x;
+  if (tid < nthreads) {
+    int outIdH = tid / outputW;
+    int outIdW = tid % outputW;
+    int inImgSize = inputW / numChannels;
+    int outImgSize = outputW / numChannels;
+    int channelId = outIdW / outImgSize;
+
+    int outImgIdy = (outIdW % outImgSize) / outImgW;
+    int inImgIdy = ratioH * outImgIdy;
+    int hId = (inImgIdy < inImgH - 1) ? 1 : 0;
+    T h1lambda = ratioH * outImgIdy - inImgIdy;
+    T h2lambda = 1.f - h1lambda;
+
+    int outImgIdx = tid % outImgW;
+    int inImgIdx = ratioW * outImgIdx;
+    int wId = (inImgIdx < inImgW - 1) ? 1 : 0;
+    T w1lambda = ratioW * outImgIdx - inImgIdx;
+    T w2lambda = 1.f - w1lambda;
+
+    T* inPos = &in[outIdH * inputW + channelId * inImgSize + inImgIdy * inImgW +
+                   inImgIdx];
+    const T* outPos = &out[outIdH * outputW + outIdW];
+    atomicAdd(&inPos[0], h2lambda * w2lambda * outPos[0]);
+    atomicAdd(&inPos[wId], h2lambda * w1lambda * outPos[0]);
+    atomicAdd(&inPos[hId * inImgW], h1lambda * w2lambda * outPos[0]);
+    atomicAdd(&inPos[hId * inImgW + wId], h1lambda * w1lambda * outPos[0]);
+  }
```

Comment:
The kernel in `.cu.h` can be moved to `.cu` file.  In addition, please note the code style.  

inImgW ->in_img_w in Fluid, this is different from old Paddle.